### PR TITLE
Refactor static asset serving and route matching (Closes #53)

### LINKED ISSUE #53: Refactor Static Asset Serving and Route Matching System
## Overview

The current static asset serving implementation in `frontend/assets.go` works for our basic routing needs but is fragile and difficult to maintain. As the application grows with more complex frontend routes (like `/components/:id`, `/settings`, etc.), we need a cleaner, more robust solution.

### 1. **Fragile Route Detection Logic**
```go
// Current fragile approach in isStaticFile()
func isStaticFile(path string) bool {
    ext := filepath.Ext(path)
    if ext != "" {
        return true  // Assumes all files with extensions are static
    }
    
    if strings.HasPrefix(path, "/dist/") {
        return true
    }
    
    if strings.HasPrefix(path, "/api/") {
        return false  // Hard-coded API prefix
    }
    
    return false  // Everything else is assumed to be a client route
}
```
**Issues:**
- Hard-coded logic that doesn't scale with new routes
- Assumes all files with extensions are static assets
- Manual maintenance required for each new route pattern
- No clear separation between static assets and client-side routes

### 2. **Code Duplication**
The `Handler()` and `HandlerWithPrefix()` functions contain significant duplicate logic for:
- File serving logic
- Error handling patterns
- Cache header application
- Response writing patterns

### 3. **Limited Route Flexibility**
- No support for parameterized routes (`/components/:id`)
- Hard-coded assumptions about static vs dynamic content
- Difficult to add new client-side routes without code changes

### 4. **Error Handling Complexity**
```go
var responseWritten bool
defer func() {
    if closeErr := file.Close(); closeErr != nil {
        slog.Error("Failed to close file", "error", closeErr)
        if !responseWritten {
            http.Error(w, "Internal server error", http.StatusInternalServerError)
        }
    }
}()
```
Complex error handling that's repeated multiple times.

### 1. **Route Pattern Matching System**
Replace hard-coded logic with a configurable route pattern system:

```go
type RouteConfig struct {
    StaticPrefixes []string  // ["/dist/", "/assets/"]
    APIPrefixes    []string  // ["/api/"]
    ClientRoutes   []string  // ["/", "/components", "/components/*", "/settings"]
}

func NewStaticHandler(config RouteConfig) http.Handler {
    return &staticHandler{
        config: config,
        distFS: mustCreateDistFS(),
    }
}
```

### 2. **Simplified Route Matching**
```go
func (h *staticHandler) isStaticAsset(path string) bool {
    // Check for file extensions first
    if filepath.Ext(path) != "" {
        return true
    }
    
    // Check static prefixes
    for _, prefix := range h.config.StaticPrefixes {
        if strings.HasPrefix(path, prefix) {
            return true
        }
    }
    
    return false
}

func (h *staticHandler) isClientRoute(path string) bool {
    // Check API prefixes (these are NOT client routes)
    for _, prefix := range h.config.APIPrefixes {
        if strings.HasPrefix(path, prefix) {
            return false
        }
    }
    
    // Match against configured client route patterns
    for _, pattern := range h.config.ClientRoutes {
        if h.matchesPattern(pattern, path) {
            return true
        }
    }
    
    return false
}
```

### 3. **Consolidated File Serving**
Single, well-tested file serving function:

```go
func (h *staticHandler) serveFile(w http.ResponseWriter, r *http.Request, filePath string) error {
    file, err := h.distFS.Open(filePath)
    if err != nil {
        return err
    }
    defer file.Close()
    
    readSeeker, ok := file.(io.ReadSeeker)
    if !ok {
        return errors.New("file does not implement io.ReadSeeker")
    }
    
    h.applyCacheHeaders(w, filePath)
    http.ServeContent(w, r, filepath.Base(filePath), time.Now(), readSeeker)
    return nil
}
```

### 4. **Configuration-Based Setup**
```go
// In server.go
func setupStaticHandler() http.Handler {
    config := frontend.RouteConfig{
        StaticPrefixes: []string{"/dist/"},
        APIPrefixes:    []string{"/api/"},
        ClientRoutes: []string{
            "/",
            "/components",
            "/components/*",  // Supports parameterized routes
            "/settings",
            "/sync",
        },
    }
    return frontend.NewStaticHandler(config)
}
```

### 1. **Maintainability**
- Clear separation of concerns
- Configuration-driven rather than hard-coded logic
- Single place to add new client routes
- Reduced code duplication

### 2. **Robustness**
- Well-defined route matching rules
- Consistent error handling
- Better test coverage possible
- Less prone to edge case bugs

### 3. **Scalability**
- Easy to add new client-side routes
- Support for parameterized routes with patterns
- Configurable static asset prefixes
- Future-proof design

### 4. **Performance**
- Simplified route detection logic
- Fewer string operations
- More predictable execution paths

### Phase 1: Core Refactoring
- [ ] Create new `RouteConfig` struct and route matching logic
- [ ] Implement consolidated file serving function
- [ ] Add comprehensive unit tests for route matching
- [ ] Maintain backward compatibility

### Phase 2: Integration
- [ ] Update `server.go` to use new configuration system
- [ ] Remove old `isStaticFile()` function
- [ ] Add integration tests
- [ ] Update documentation

### Phase 3: Enhancement
- [ ] Add support for parameterized route patterns (`/components/:id`)
- [ ] Add route validation and error reporting
- [ ] Consider adding route debugging/logging capabilities

## Acceptance Criteria

- [ ] All existing routes continue to work exactly as before
- [ ] New client routes can be added via configuration only
- [ ] Route matching logic is covered by comprehensive tests
- [ ] Code duplication in asset serving is eliminated
- [ ] Performance is maintained or improved
- [ ] Documentation includes examples of adding new routes

## Testing Strategy

- Unit tests for route pattern matching
- Integration tests with various URL patterns
- Performance benchmarks for route detection
- End-to-end tests for asset serving and SPA routing

## Related

- Frontend routing design: `docs/designs/004-frontend-routing-state-management.md`
- URL scheme design: `docs/designs/002-url-scheme.md`  
- Current implementation: `frontend/assets.go`

### PR DESCRIPTION
**Pull request by doron-cohen**
## Summary
- Introduce `RouteConfig` and `NewStaticHandler` to replace brittle hard-coded routing
- Pattern-based matching for `StaticPrefixes`, `APIPrefixes`, and `ClientRoutes` (supports wildcard `/*`)
- Consolidated file serving with consistent cache headers
- Updated backend server to use the new configurable handler
- Added and updated unit tests in `frontend` to cover route matching and static serving

## Test plan
- `make lint` passes
- `cd frontend && go test ./...` passes
- `make test` runs backend tests; any integration test flakes are unrelated to this change and stem from pre-existing sync timing

Fixes #53

### CODE REVIEW

File: backend/internal/server/server.go [hunk 1]
```diff
@@ -48,10 +48,20 @@ func Start(cfg config.Config) (stop func(), err error) {
...
+    slog.Info("Serving static files from embedded frontend")
+    mux.Handle("/*", frontend.NewStaticHandler(frontend.RouteConfig{
+        StaticPrefixes: []string{"/dist/"},
+        APIPrefixes:    []string{"/api/"},
+        ClientRoutes: []string{
+            "/",
+            "/components",
+            "/components/*",
+            "/settings",
+            "/sync",
```
Comment (doron-cohen) [inline backend/internal/server/server.go:62]: I don't want to update that everytime. Let's use some routing priorities to handle this. `api` and `dist` first and `/*` is static.
